etna/analysis/eda_utils.py

```diff
@@ -194,25 +197,69 @@ def sample_acf_plot(
     figsize:
         size of the figure per subplot with one segment in inches
     """
+    nan_flag_segments = False
     if segments is None:
+        nan_flag_segments = True
         segments = sorted(ts.segments)
 
+    plot = plot_pacf if partial else plot_acf
+    title = "Partial Autocorrelation" if partial else "Autocorrelation"
+
     k = min(n_segments, len(segments))
-    columns_num = min(2, k)
-    rows_num = math.ceil(k / columns_num)
+    fig, ax = prepare_axes(num_plots=k, columns_num=columns_num, figsize=figsize)
+    fig.suptitle(title, fontsize=16)
 
-    figsize = (figsize[0] * columns_num, figsize[1] * rows_num)
-    fig, ax = plt.subplots(rows_num, columns_num, figsize=figsize, constrained_layout=True, squeeze=False)
-    ax = ax.ravel()
-    fig.suptitle("Autocorrelation", fontsize=16)
-    for i, name in enumerate(sorted(np.random.choice(segments, size=k, replace=False))):
+    if nan_flag_segments:
+        used_segments = sorted(np.random.choice(segments, size=k, replace=False))
+    else:
+        used_segments = segments
+    for i, name in enumerate(used_segments):
         df_slice = ts[:, name, :][name]
-        plot_acf(x=df_slice["target"].values, ax=ax[i], lags=lags)
+
+        if df_slice["target"].isna().any():
+            print("yes")
+            df_slice["target"].dropna(inplace=True)
+            warnings.warn("Values with NaN dropped!")
```

Comment:
I don't think we should raise this warning. Let's clarify this moment in docstring of the function. Smth like:
```
This function removes any NaNs before plotting.
```